Fix/yuki punishment edging

DiligentJackal opened the pull request:
Added additional question asking if the player orgasmed or not as a check, still no puppy status variable yet

DiligentJackal committed:
More line rewrites and organization

Huge rewrite of lines, reorg of unrelated lines to the bottom of file temporarily, to eventually be moved elsewhere or discarded.

DiligentJackal committed:
Add additional question for Yuki_Punishment_Edging

Added additional question asking if the player orgasmed or not as a check, still no puppy status variable yet

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -261,13 +259,22 @@
-    -> Yuki_AcknowledgeNo
+    -> Yuki_AcknowledgeNo -> 
+    
+    //change to a shuffle and add lines
```
Do you need help to make a shuffle or you want someone else to do it?

DiligentJackal replied:
Ah hah! Lost this one, should've been a TODO instead of a comment, thank you for flagging! I'll get it changed in the next commit.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -19,7 +19,6 @@
     Do you accept being my puppy for this punishment? 
```
"Now, before you get started working off that Obsidian..."

We usually capitalize Bloodstone and Obsidian. And I often use ellipses to create a dramatic pause.

DiligentJackal replied:
Fixed! I will remember that!

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -19,7 +19,6 @@
 
-
 + [pose:puppy_yes]
     I'm glad you're starting to behave, puppy! 
```
I'm not sure they are behaving at this point.

Suggest something like: "Good. I'm glad you know your place."

Following on from that, the next sentence probably wouldn't start with 'then'.

Suggest: "Time to sit down and listen."

DiligentJackal replied:
Hm, good point, agreed and adjusted! :)

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -32,7 +31,7 @@
     
```
Nice. I really like this little chastisement for acting like a puppy without agreeing :)

DiligentJackal replied:
<3 Thanks! :D

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -32,7 +31,7 @@
+    -> puppy_check
 
 + [Yes]
     I'm glad you're starting to behave, puppy! 
```
The player is saying yes, but not in 'puppy' mode.

Maybe it would be better to correct them and say something like:

"Well, if that's the case, then answer with your paws.
Try again, {player_name}."

Then you can remove the next line "Then, sit and listen."

DiligentJackal replied:
I was being too nice haha, agreed, adjusted to your suggestion and added -> to go back to the question again. :) 

Thanks!

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -43,6 +42,8 @@
 
 + [No]
```
"Oh, what am I going to do with you?"

Just some punctuation corrections.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -64,13 +65,12 @@
 - (explained_rules)
 /perform talk_consider skeptical 70
-  But I don't quite understand <i>why</i> you need to be punished.
+  But I don't quite understand <i>why</i> you misbehaved.
```
Maybe it would sound better to say, "What I don't understand is why you misbehaved."

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -89,14 +89,14 @@
 
   /perform react_point_at_player_angry angry 80
-    For your total lack of respect towards me, and this house;
+    For your total lack of respect towards me and this house,
```
We always capitalize House.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -89,14 +89,14 @@
+    For your total lack of respect towards me and this house,
 
   /perform do_tap_wrist angry 70
   ~ addPunishment()
```
punishment marker

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -111,9 +111,6 @@
-
     
 -
 /perform do_point_down skeptical 65
```
Suggest making the second part of the sentence a new sentence:

"Think about what you've done."

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -127,7 +124,7 @@
 /perform idle_unhappy amused 55
  //add shuffle of lines
```
What kind of shuffle of lines would it be?

Not sure if this is what you meant, but the player needs to be told to grovel and stay in the pose until you tell them they can move. That's a way for the game to tell them what to do, even if we don't have the functionality to make sure they are holding the pose yet.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -127,7 +124,7 @@
+TODO Could add a longer check here to keep the player groveling, should make it feel more like a punishment
 
 +[pose:grovel]
 +{ tries < 2 } [timeout] -> grovelamusement
```
The 'tries' is old code. We find that it's not really necessary. You can skip them and just do a normal timeout.

Might be good to have some lines in here that Yuki says while you're still in the grovel pose.

You could use the same random comment loop from line 253 later in the scene. You just might need to create a new set of comments.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -207,40 +204,41 @@
 /perform idle_tease aroused 55
-  For your punishment today, you're going to edge for me, but you're not going to be allowed to cum. 
-
+  For your punishment today, you're going to touch yourself for me, until you're aching, 
```
Just a minor formatting feedback. Our subtitle system doesn't work well with long text lines. It's very messy on screen.

So, a good practice is to break them up or wrap them where you can. It's also good to remember that when I'm acting, each line is treated as it's own audio file and tagged by the system. That means I need a good natural break point to take a breath or pause so the audio can be spliced.

For these lines, I would suggest this:

"For your punishment today,
you're going to touch yourself for me
unitl you're aching.

And then...
Even though it's going to pain me,
I'll have to leave you wanting this time."

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -207,40 +204,41 @@
-    /perform HAPPY
+    TODO set perform
+    /perform HAPPY 
     Good pup! Now get those paws busy on your {hasPenis: cock|pussy}!
```
Do you want it to be one paw or two?

If just one, you could instruct them to keep the other paw on their head.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -207,40 +204,41 @@
-    
+    TODO set perform
     /perform ANNOYED 
     Oh puppy... You have to take your punishment somehow...
```
Sometimes, once a scene has started, if a player says no to what they are supposed to do, we double check to make sure they want to say no. 

You simply ask them, "Are you sure you want to stop?"

And the formatting for the yes / no answers is ++[yes] and ++[no]. The double plus marks denotes that the system should check for those responses _after_ it checks for the first set of yes / no answers.

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -261,13 +259,22 @@
   /perform react_laugh amused 60
   ~ addObey()
-    You're not supposed to enjoy it thaaat much!
+    Goodie! But remember, no cumming.
```
I tend to break short statements like these into two lines, because it read better with the VO when I record it.

I will say "Goodie!" then pause, and then say "But remember... no cumming!"

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -261,13 +259,22 @@
+    
+    Good, you're being punished after all. 
+    
+    -> edging_break_1
```
Does this line play after "no" or after either "yes" or "no"?

If it's after both, it needs two dashes. The -> sends it to a new place in the script, but just a single dash tells Ink to continue the story after the player answers. So --> edging_break_1 will make sure the dialogue continues and it redirects the player.

If it's after just 'no', then the way you have it is fine :)

DiligentJackal replied:
Yes, my intention is to add a shuffle specifically after the player says 'no' here, and the shuffle will currently contain the three possible lines: 
"Well maybe next time you shouldn't be such a naughty little puppy, hmmm?", 
"Maybe you should move those paws a little better then?", and 
"Good, you're being punished after all." 

It got flubbed because I lost my intention to create a shuffle here by commenting it instead of TODOing it. :) I'll adjust this to a properly coded shuffle, thank you!

DiligentJackal replied:
Properly coded in this shuffle that plays after a 'no' answer, resolving. :)

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -313,10 +321,10 @@
 /perform react_chuckle amused 55
-  Hehehe, I love making my puppies stand up on their hind legs;
-  it's very entertaining, like you're a person.
+  Hehehe, I love making my puppies stand up on their hind legs.
```
Suggest breaking this into two lines:

"Hehehe.
I love making my puppies stand on their hind legs."

DiligentJackal replied:
Done!

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -313,10 +321,10 @@
-  Hehehe, I love making my puppies stand up on their hind legs;
-  it's very entertaining, like you're a person.
+  Hehehe, I love making my puppies stand up on their hind legs.
+  It's very entertaining, like you're a person.
```
"It's very entertaining.
Like you're a person!"

DiligentJackal replied:
Fixed, ez! :D

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -313,10 +321,10 @@
+  But remember, you're just my puppy right now and you're being punished. 
 
 /perform idle_tease aroused 45
   Let's continue with your punishment.
```
You'll want to tell them to get out of the standing position and into some other position.

Unless you mean to have them masturbate standing? If so, you should still tell them what pose to be in for the next round. :)

DiligentJackal replied:
Gooooood catch, thank you! I will think about this and fix it in the next commit. 

 Related, I'd also love to get a reference sheet of possible poses and what they look like so I know what I can call on, that'd actually be a colossal help for me when writing scenes and envisioning it from the player's perspective.

DiligentJackal replied:
Adjusted to sitting :)

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -292,6 +299,7 @@
 
 /perform do_summon_player excited 35
+TODO change to player name
   Stand up for me. Like a good little {isBoy: boy| {isGirl: girl| pet}}.
```
"Stand up for me.
Like a good little {player_name}."

I just realized some of my comments are on the stuff you didn't write. Sorry about that!

DiligentJackal replied:
That's okay, regardless of who created it, it needs to be fixed now :) 

Fixed this.

DiligentJackal committed:
Incorporating some but not all feedback (yet)

Starting to resolve various things from Domina and Taco's feedback today :)

DiligentJackal committed:
Adding more direct instructional dialogue

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -258,21 +262,20 @@ TODO Could add a longer check here to keep the player groveling, should make it
   /perform react_laugh amused 60
   ~ addObey()
-    Goodie! But remember, no cumming.
+    Good! But remember, no matter how good it feels, you do NOT have permission to cum.
```
I actually like how the line was before. I know you're explaining more, but the short clipped childish lines are more in keeping with Yuki's youthful enthusiasm. :)

DiligentJackal replied:
Reverted :)

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -451,13 +476,13 @@ TODO set perform
   I am a kind and generous mistress, and even though this is still a punishment, I want to see my good puppy make the most of it. 
   
-  I can see you need it. 
+  I can see you want the pleasure. 
```
Maybe break up the line starting with "I am a kind and generous Mistress..." so that it will show better in subtitles in game.

For the second line, it's too formal and Yuki is more casual and enthusiastic. Perhaps try, "I know how much you really to."

DiligentJackal replied:
Broken up that long run-on line now, and I decided to actually just remove that other line all together, it feels better to me that way

DevilishDomina commented on ink/yuki/punishment_edging:
```diff
@@ -827,6 +851,36 @@ TODO set perform animations
 
 ->->
 
+TODO add call to tease_bank_joi in each edging segment and add more lines here
```
I like where you're going with this. I want to make sure the scene doesn't feel like just another JOI scene, but without the ability to cum, so try to make sure the new lines aren't instructions. Maybe they can be comments on their performance or how much she's looking forward to playing with them as a good puppy.

DiligentJackal replied:
Edited, commented out more of the JOI style lines. :) Appreciate the feedback here.

DiligentJackal committed:
incorporating more comments

Working off most recent round of feedback, still have more to do around the "joi" bank addition in each section

DiligentJackal committed:
Implemented poses and tease_bank of addtl lines

tease_bank_joi now live in relevant sections, poses adjusted to incorporate feedback.

I think this is ready for a final look. :)
